cmd/wgcf-cli/cmd_generate.go

```diff
@@ -13,64 +16,204 @@ import (
 
 var generateCmd = &cobra.Command{
 	Use:       "generate",
-	Short:     "Generate a xray/sing-box config",
+	Short:     "Generate a xray/sing-box/wg-quick config",
 	Run:       generate,
 	Args:      cobra.OnlyValidArgs,
-	ValidArgs: []string{"--xray", "--sing-box", "--wg"},
+	ValidArgs: []string{"--xray", "--xray-module", "--xray-tag", "--xray-indent-width", "--sing-box", "--wg", "--wg-quick", "--output-file"},
+}
+
+type OutputFileType int8
+
+const (
+	Stdout OutputFileType = iota
+	Default
+	Custom
+)
+
+type GeneratorType int8
```

Comment:
There is functions which require current generator type (main_test.go). Also see https://github.com/ArchiveNetwork/wgcf-cli/pull/34#discussion_r1792050515. 

Boolean value not fit in this case, because we need N function arguments in case N generators. Instead of usage raw numeric value better to use enumeration because of better readability and same or lesser memory overhead.

If we need to support multiple generators, then enumerator values just can be set to power of 2 values (1,2,4,8...), so in uint8 we can store up to 8 generators simultaneously, as example.